File: 0050-powx-n/0050-powx-n.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n == 0: return 1
        if n < 0: return 1.0/self.myPow(x, -n)
        tmp = self.myPow(x, n//2)
        return tmp * tmp if n % 2 == 0 else tmp * tmp * x

File: 0050-powx-n/0050-powx-n.cpp
class Solution {
public:
    double myPow(double x, int n) {
        if (n == 0) return 1;
        if (n == INT_MIN) return 1.0/(x * myPow(x, -(n + 1))); //-INT_MIN would cause an overflow error
        if (n < 0) return 1/myPow(x, -n);
        double tmp = myPow(x, n/2);
        if (n % 2 == 0) return tmp * tmp;
        return tmp * tmp * x;
    }
};